content/blog/2018-07-17-pkginspector.md

```diff
@@ -0,0 +1,112 @@
+
+*What's not working:* We know that we miss function calls if they are passed as parameters to `purrr::map()` and `do.call()` functions. There may be other systematic misses as well.
+
+### Visualization
+
+`vis_package()` helps you visualize the network of functions' dependencies ([interactive example](http://rpubs.com/jtr13/vis_package)).
+
+
+```r
+vis_package(path, physics = FALSE)
+```
+
+![](img/blog-images/2018-07-17-pkginspector/viridisLite.png)
+
+### Argument default usage
+
+`rev_args()` identifies all the functions' arguments used in a given package. It returns a dataframe which main column, `detault_consistent` indicates whether or not the default value of the argument is consistent across the functions that use it. This helps to evaluate the complexity of the package and to identify potential sources of confusion, for example, if the meaning or default value of the same argument varies across functions.
+
+
+```r
+rev_args(path)$arg_df
+```
+
+```
+##    arg_name n_functions default_consistent default_consistent_percent
+## 1         n           6              FALSE                   83.33333
+## 2     alpha           6               TRUE                  100.00000
+## 3     begin           6               TRUE                  100.00000
+## 4       end           6               TRUE                  100.00000
+## 5 direction           6               TRUE                  100.00000
+## 6    option           2               TRUE                  100.00000
+```
+
+The example shows that the parameter `n` is used inconsistently. [The documentation](https://github.com/sjmgarnier/viridisLite/blob/master/R/viridis.R) reveals that the default value of `n` is 256 in one function but missing in all others. This flags a potential issue that deserves further investigation. In this case, the odd function out--`viridisMap()`--has a clear use case that is different from the others.
+
+### In sum
+
+If you are building or reviewing an R package, __pkginspector__ can help you better understand its complex structure. This is an important step towards improving your code and research. While __pkginspector__ will expand in scope, the features built during and since unconf18 are already useful. We welcome comments and issues about what other elements of a package deserve to be reviewed and if that review process can be automated. Our broader vision for __pkginspector__ is a tool that guides both the development and review of R packages and provide automated checks on subtle differences in package functions that inevitably arise during the development process. The package will (hopefully) grow and exist as a living toolbox for development and review. 
```

Comment:
in "We welcome comments and issues", consider linking to where you want someone to open an issue (https://github.com/ropenscilabs/pkgreviewr/issues?) and maybe even be more explicit about saying "...encourage you to open an issue" or similar. 

In the case of opening an issue, does it make it confusing that the repo is `pkgreviewr` so does this need a one-line explanation of their relationship somewhere?

Reply:
The repo is actually: https://github.com/ropenscilabs/pkginspector. @stefaniebutland this repo location has changed to unconf. 

> We welcome comments and issues about what other elements of a package deserve to be reviewed and if that review process can be automated.

to 

We encourage you to open an [issue](https://github.com/ropenscilabs/pkginspector/issues) about what other elements of a package deserve to be reviewed especially if that review process can be automated.